extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/error/MSQWarningReportLimiterPublisher.java

```diff
@@ -108,6 +115,12 @@ public void publishException(int stageNumber, Throwable e)
         return;
       }
     }
+    if (CannotParseExternalDataFault.CODE.equals(errorCode)) {
```

Comment:
This is what I would do: 
Rename `MSQWarningReportPublisher` to `ExceptionPublisher`.


Create a new CompositeExceptionPublisher.

One implementation can be of a FilteredEmitterExceptionPublisher where in the constructor you take the error code. 

It would be cool to avoid This call
:MSQErrorReport msqFault = MSQErrorReport.fromException(workerId, host, stageNumber, e);

Reply:
`:MSQErrorReport msqFault = MSQErrorReport.fromException(workerId, host, stageNumber, e);`
will definitely avoid repeating this call

Reply:
How do i go about attaching a debugger in MSQ to see if these reports are being captured/emitted

Reply:
I'm not sure i follow 

> This is what I would do:
> Rename MSQWarningReportPublisher to ExceptionPublisher.
> 
> Create a new CompositeExceptionPublisher.
> 
> One implementation can be of a FilteredEmitterExceptionPublisher where in the constructor you take the error code.

Reply:
> How do i go about attaching a debugger in MSQ to see if these reports are being captured/emitted

MSQWarningsTest#testThrowExceptionWhenParseExceptionsExceedLimit should be a good one. 

`CompositeExceptionPublisher` will implement the renamed interface `ExceptionPublisher` something like

```java
class CompositeExceptionPublisher implements ExceptionPublisher{
List<ExceptionPublisher> delegates;

public CompositeExceptionPublisher(List<ExceptionPublisher> delegates){

this.delegates=delegates;
}

void publishException(int stageNumber, Throwable e){
for(ExceptionPublisher delegate:delegates){
delegate.publishException(stage,e);
}
}


  void close() throws IOException{
for(ExceptionPublisher delegate:delegates){
delegate.close();
}
}


}

```

Now you can create a new impl `FilteredEmitterExceptionPublisher` and pass it errorCodes in the constructor. 

Then in WorkerImpl#runTask method create a CompositeExceptionPublisher with MSQWarningReportLimiterPublisher and FilteredEmitterExceptionPublisher and pass it along. 

Maybe this strategy can work.

Reply:
@cryptoe How does this look?
We have the Limiter publisher and we pass in the composite publisher that contains the normal publisher and the emitter publisher